src/FeatureUtil/FeatureUtil.js

```diff
@@ -22,7 +23,29 @@ class FeatureUtil {
     let featureId = feature.getId();
     let featureIdParts = featureId ? featureId.split('.') : featureId;
 
-    return Array.isArray(featureIdParts) ? featureIdParts[0] : undefined;
+    return isArray(featureIdParts) ? featureIdParts[0] : undefined;
+  }
+
+  /**
+   * Extracts the featureType name from given GetFeatureInfo URL.
+   * This method is mostly useful for raster layers which features could have
+   * no ID set.
+   *
+   * @param {string} url GetFeatureInfo URL possibly containing featureType name.
+   *
+   * @return {string} Obtained featureType name as string.
+   */
+  static getFeatureTypeNameFromGetFeatureInfoUrl(url) {
+    const regex = /query_layers=(.*?)&/i;
+    let match = url.match(regex);
+    let featureTypeName;
+    if (isArray(match) && match[1]) {
+      featureTypeName = decodeURIComponent(match[1]);
+      if (featureTypeName.indexOf(':') > 0) {
+        featureTypeName = featureTypeName.split(':')[1];
```

Comment:
Why do we always unqualify? This is sth. that application code should do or that should at least be configurable.